Src/tim.c

```diff
@@ -53,9 +53,13 @@ void MX_TIM16_Init(void)
   TIM_BreakDeadTimeConfigTypeDef sBreakDeadTimeConfig;
 
   htim16.Instance = TIM16;
-  htim16.Init.Prescaler = 0;
+
+  // 16bit timer is 65535 counts. Desired period is 20ms. Clock frequency is 48MHz
+  // With a prescaler of 32 the period would be 65535/(48M/32) = 43.69ms.
+  // With the period being set to (65535)/2, the period would be 43.69/2 = 21.845ms
+  htim16.Init.Prescaler = 16;
   htim16.Init.CounterMode = TIM_COUNTERMODE_UP;
-  htim16.Init.Period = 0;
+  htim16.Init.Period = (65535)/2;   
```

Comment:
Can you change this value so the period is as close to 20 ms as possible? Note that this can be any value between 0 - 2^16-1